pandas/_libs/tslib.pyx

```diff
@@ -95,7 +95,7 @@ def _test_parse_iso8601(ts: str):
         return Timestamp.now().normalize()
 
     string_to_dts(ts, &obj.dts, &out_bestunit, &out_local, &out_tzoffset, True)
-    obj.value = dtstruct_to_dt64(&obj.dts)
+    obj.value = npy_datetimestruct_to_datetime(NPY_FR_ns, &obj.dts)
```

Comment:
FWIW: It looks like this entire function is just unit tested and not used anywhere else so could be removed?

Reply:
yah, i always figured it was really a test for string_to_dts, would be OK with declaring we already have enough indirect tests of that